lang/cs/Org.Apache.REEF.Driver/Bridge/Events/FailedEvaluator.cs

```diff
@@ -39,6 +43,22 @@ public FailedEvaluator(IFailedEvaluatorClr2Java clr2Java)
             _failedContexts = new List<IFailedContext>(
                 FailedEvaluatorClr2Java.GetFailedContextsClr2Java().Select(clr2JavaFailedContext => 
                     new FailedContext(clr2JavaFailedContext)));
+
+            var errorBytes = FailedEvaluatorClr2Java.GetErrorBytes();
+            if (errorBytes != null)
+            {
+                var formatter = new BinaryFormatter();
+                using (var memStream = new MemoryStream(errorBytes))
+                {
+                    var inner = (Exception)formatter.Deserialize(memStream);
+                    _evaluatorException = new EvaluatorException(_id, inner.Message, inner);
+                }
+            }
+            else
+            {
+                _evaluatorException = new EvaluatorException(
```

Comment:
No, this is for when the Exception originates from Java. I've added a comment in my next iteration.